### DISCUSSION TITLE
Is there anyway we can get the (approximate) time/frame of the ouput

### DISCUSSION THREAD

**Discussion by samshipengs** [snakers4/silero-models#33] [Q&A] [answered]
Given a clip of 5s speech, is there anyway to get the rough time when each word was spoken?
e.g. the output might look like this
`"hello world here is a question" ` 
 `0s     2s       3s    4s  4.5s  5s`
(or, the frame numbers from input waveform)

**Comment by snakers4** [comment, marked as answer]
There is an example for PyTorch producing timestamps in the colab

**Comment by samshipengs** [comment]
I see, thanks for pointing out. By turn on the `word_align` flag in `decoder` there will be time alignment outputs. But it seems like the timing of each word is all the same i.e. `wav_len / len(argm)` I think, wouldn't there be issue for prolonged pronunciation (someone saying a word very slowly)? and another question is, I didn't quite understand how the `decoder` function code calculates the beginning time of each word, would you mind sharing some high level intuition on that? (sorry im a newbie to audios...)

**Comment by snakers4** [comment]
In this particular case basic lm is integrated into the model so it knows when words start and end

Longer pronunciation to a reasonable extent should not be a problem since this was accounted for during training
